File: switch-cash atm/atm.java
import java.util.Scanner;

public class atm {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        boolean valid = false;
        String nama;

        do {
            System.out.print("Masukkan Nama: ");
            nama = input.nextLine();

            System.out.print("Apakah nama sudah benar? (true/false): ");
            String cek = input.nextLine();

            if(cek.equalsIgnoreCase("true")) valid = true;
            else System.out.println("Masukkan nama kembali!\n");
        } while(!valid);

        System.out.print("Masukkan NIM (angka): ");
        String nim = input.nextLine();
        int saldo = Integer.parseInt(nim);

        System.out.println("\n===== DATA AKUN =====");
        System.out.println("Nama : " + nama);
        System.out.println("Saldo Awal : Rp " + saldo);

        int pilih;
        do {
            System.out.println("\n===== MENU ATM =====");
            System.out.println("1. Cek Saldo\n2. Tarik Tunai\n3. Setor Tunai\n4. Transfer\n5. Keluar");
            System.out.print("Pilih menu: ");
            pilih = input.nextInt();

            switch(pilih){
                case 1 -> System.out.println("Saldo Anda: Rp " + saldo);
                case 2 -> { System.out.print("Nominal tarik: "); saldo -= input.nextInt(); }
                case 3 -> { System.out.print("Nominal setor: "); saldo += input.nextInt(); }
                case 4 -> { System.out.print("Nominal transfer: "); saldo -= input.nextInt(); }
            }

        } while(pilih != 5);

        System.out.println("Terimakasih menggunakan ATM!");
    }
}

File: switch-cash atm/atm.cpp
#include <iostream>
using namespace std;

int main() {
    string nama;
    string cek;
    bool valid = false;
    long saldo;
    string nim;

    while(!valid)
        cout << "Masukkan Nama: ";
        getline(cin, nama);

        cout << "Apakah nama sudah benar? (true/false): ";
        cin >> cek;
        cin.ignore();

        if(cek == "true") valid = true;
        else cout << "Masukkan nama kembali!\n\n";
    }

    cout << "Masukkan NIM (angka): ";
    cin >> nim;
    saldo = stol(nim);

    int pilih;
    do {
        cout << "\n===== MENU ATM =====\n";
        cout << "1. Cek Saldo\n2. Tarik Tunai\n3. Setor Tunai\n4. Transfer\n5. Keluar\n";
        cout << "Pilih menu: ";
        cin >> pilih;

        switch(pilih){
            case 1:
                cout << "Saldo Anda: Rp " << saldo << endl;
                break;
            case 2:
                long tarik;
                cout << "Nominal tarik: ";
                cin >> tarik;
                saldo -= tarik;
                break;
            case 3:
                long setor;
                cout << "Nominal setor: ";
                cin >> setor;
                saldo += setor;
                break;
            case 4:
                long transfer;
                cout << "Nominal transfer: ";
                cin >> transfer;
                saldo -= transfer;
                break;
        }
    } while(pilih != 5);

    cout << "Terimakasih menggunakan ATM!";
}